base-env.lua

```diff
@@ -10,6 +10,8 @@ local evaluator = require "./evaluator"
 
 local p = require "pretty-print".prettyPrint
 
+---@alias lua_operative fun(syntax : any, env : Environment, goal : ExpressionGoal) : boolean, any, Environment | nil
```

Comment:
Please make this type stricter; the second return value can be made more precise

Reply:
Can't without significantly increasing scope, tried to document intent better but the type is still not constrained because it has to allow unknown for errors.